pulp_rpm/tests/functional/conftest.py

```diff
@@ -147,10 +187,16 @@ def _init_and_sync(
 @pytest.fixture(scope="class")
 def rpm_unsigned_repo_immediate(init_and_sync):
     repo, _ = init_and_sync()
-    yield repo
+    return repo
 
 
 @pytest.fixture(scope="class")
 def rpm_unsigned_repo_on_demand(init_and_sync):
     repo, _ = init_and_sync(policy="on_demand")
-    yield repo
+    return repo
+
+
+@pytest.fixture(scope="class")
```

Comment:
I don't suppose there's any way to have this parameterized while still getting the "only do it once" effects?  Perhaps even doing it ourselves with something like functools.cache? https://docs.python.org/3/library/functools.html#functools.cache

Reply:
Or maybe we cannot do that because some tests are going to end up modifying those repos.

Reply:
Yep, exactly. These should be used only for class-scoped tests that do not manipulate repositories.